vendors/nordic/boards/nrf52840-dk/ports/ota_pal_for_aws/ota_pal.c

```diff
@@ -0,0 +1,734 @@
+ */
+static ret_code_t otaPal_WriteImageDescriptor( OtaFileContext_t * const pFileContext );
+
+/**
+ * @brief Sincronously write data to flash
+ *
+ * @param[in] ulOffset Position to write
+ * @param[in] ulBlockSize Data size
+ * @param[in] pacData Data to be written
+ *
+ * @return Amount of written data or -1 in the case of error
+ */
+static ret_code_t prvWriteFlash( uint32_t ulOffset,
+                                 uint32_t ulBlockSize,
+                                 uint8_t * const pacData );
+
+static int prvNextFreeFileHandle = 1;
+static EventGroupHandle_t xFlashEventGrp = NULL;
+
+/*-----------------------------------------------------------*/
+
+ret_code_t prvComputeSHA256Hash( void * pvMessage,
+                                 size_t ulMessageSize,
+                                 void * pvHashDigest,
+                                 size_t * pulDigestSize );
+
+/*-----------------------------------------------------------*/
+
+ret_code_t prvVerifySignature( uint8_t * pusHash,
+                               size_t ulHashSize,
+                               uint8_t * pusSignature,
+                               size_t ulSignatureSize,
+                               uint8_t * pucPublicKey,
+                               uint32_t ulPublicKeySize );
+
+/*-----------------------------------------------------------*/
+
+OtaPalMainStatus_t prvErasePages( size_t xFrom,
+                                  size_t xTo )
+{
+    size_t ulErasedAddress = ( size_t ) xFrom;
+    OtaPalMainStatus_t xStatus = OtaPalSuccess;
+    ret_code_t xErrCode = NRF_SUCCESS;
+
+    while( xStatus == OtaPalSuccess && ulErasedAddress < ( size_t ) xTo )
+    {
+        size_t ulErasedPage = ( size_t ) ( ulErasedAddress / NRF_FICR->CODEPAGESIZE );
+        ulErasedAddress += NRF_FICR->CODEPAGESIZE;
+        xEventGroupClearBits( xFlashEventGrp, otapalFLASH_SUCCESS | otapalFLASH_FAILURE );
+        xErrCode = sd_flash_page_erase( ulErasedPage );
+
+        if( xErrCode != NRF_SUCCESS )
+        {
+            xStatus = OtaPalRxFileCreateFailed;
+        }
+        else
+        {
+            /**
+             * If soft device is enabled, the result of sd_flash_page_erase() is posted via event, otherwise sd_flash_page_erase()
+             * is a blocking operation.
+             */
+            if( nrf_sdh_is_enabled() )
+            {
+                EventBits_t xFlashResult = xEventGroupWaitBits( xFlashEventGrp, otapalFLASH_SUCCESS | otapalFLASH_FAILURE, pdTRUE, pdFALSE, portMAX_DELAY );
+
+                if( xFlashResult & otapalFLASH_FAILURE )
+                {
+                    xStatus = OtaPalRxFileCreateFailed;
+                }
+            }
+        }
+    }
+
+    return xStatus;
+}
+
+OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const pFileContext )
+{
+    ret_code_t xErrCode;
+
+    pFileContext->pFile = prvNextFreeFileHandle;
+    OtaPalMainStatus_t xStatus = OtaPalSuccess;
```

Comment:
Can you swap these lines so that we aren't mixing declarations with functional statements?